### PR TITLE
Add options for empty-layer history entries

### PR DESCRIPTION
**Pull request by nalind**
Add configuration methods for adding entries which will show up in a committed image's history, both before and after the new layer that we add while committing the image.  Expose them from the CLI in the form of a new --add-history option for the "add", "config", "copy", and "run" commands.

### CODE REVIEW

File: cmd/buildah/config.go [hunk 1]
```diff
@@ -17,6 +18,10 @@ import (
 
 var (
 	configFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "add-history",
```
Comment (rhatdan) [inline cmd/buildah/config.go]: Lets add an environment variable to turn this on or off by default.

Comment (nalind) [reply cmd/buildah/config.go]: Added `$BUILDAH_HISTORY` for it.

File: docs/buildah-add.md [hunk 2]
```diff
@@ -15,6 +15,14 @@ archive file itself.  If a local directory is specified as a source, its
...
 ## OPTIONS
 
+**--add-history**
+
+Add an entry to the history which will note the digest of the added content.
+Defaults to false.
+
+Note: You can also override the default value of --add-history by setting the
+BUILDAH\_HISTORY environment variable. `export BUILDAH_HISTORY=true`
+
```
Comment (TomSweeneyRedHat) [inline docs/buildah-add.md]: :+1:

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**